app/src/main/kotlin/compiler/analysis/NameResolver.kt

```diff
@@ -196,7 +204,9 @@ object NameResolver {
 
                 is Expression.FunctionCall -> {
                     if (!checkFunctionUsage(node)) {
-                        nameDefinitions[Ref(node)] = Ref(visibleNames[node.name]!!.topFunction())
+                        // if this is a generator call then analysis of Statement.ForEachLoop has already resolved this name
```

Comment:
Also please watch out there for incorrectly calling generator as it was a normal function, since previous phases don't and cannot detect it.

Reply:
Now I changed it so this case will happen only if this is a regular function call.
Generator calls can only occur in `ForEachLoop` which will perform name resolution and **won't** recursively analyse the `generatorCall`, but directly analyse its arguments.